lucene/core/src/java/org/apache/lucene/search/IndriScorer.java

```diff
@@ -0,0 +1,37 @@
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+/**
+ * The Indri parent scorer that stores the boost so that 
+ * IndriScorers can use the boost outside of the term.
+ *
+ */
+abstract public class IndriScorer extends Scorer {
+
+	private float boost;
```

Comment:
Hmm is this (to store `boost`) the only reason to have a separate `IndriScorer`?  If I remember right, Lucene used to apply `boost` similarly (every scorer kept track of it) but at one point we moved all boosting to a dedicated `BoostQuery`.

Reply:
I did this because I apply the boost in the scorer rather than in the similarity (such as in LMDirichletSimilarity), and I divide by the sum of the boosts.  I originally did this to exactly match the Indri scores; however, this is not a huge priority.  I don't have much issue with dropping this if it doesn't fit in the Lucene workflow well.